spyder/plugins/editor/lsp/transport/main.py

```diff
@@ -109,16 +113,19 @@ def restore(self):
 
 if __name__ == '__main__':
     logger_init(args.transport_debug)
+    unknownargs = [x for x in unknownargs if len(x) > 0]
+    unknownargs = ' '.join(unknownargs)
+    logger.debug(unknownargs)
     process = psutil.Process()
     sig_manager = SignalManager()
-    client = LanguageServerClient(host=args.server_host,
-                                  port=args.server_port,
-                                  workspace=args.folder,
-                                  zmq_in_port=args.zmq_in_port,
-                                  zmq_out_port=args.zmq_out_port,
-                                  use_external_server=args.external_server,
-                                  server=args.server,
-                                  server_args=unknownargs)
+    LanguageServerClient = partial(TCPLanguageServerClient,
+                                   host=args.server_host,
+                                   port=args.server_port)
+    if args.stdio_server:
+        LanguageServerClient = partial(StdioLanguageServerClient,
+                                       server_args=unknownargs)
```

Comment:
Is it possible to write this like:

```python
if args.stdio_server:
    LanguageServerClient = partial(StdioLanguageServerClient,
                                   server_args=unknownargs)
else:
    LanguageServerClient = partial(TCPLanguageServerClient,
                                   host=args.server_host,
                                   port=args.server_port)
```

?

I think it'd be clearer this way.